File: 0028.实现strStr()/0028-实现strStr().cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        if (needle.size() == 0){
            return 0;
        }
        int lh = haystack.size();
        int ln = needle.size();
        bool flag = 0;
        for (int i = 0; i <=lh - ln ; ++i){
            flag = 1;
            for (int j = i; j < i + ln; ++ j){
                if (haystack[j] != needle[j-i]){
                    flag = 0;
                    break;
                }
            }
            if (flag) return i;
        }
        if (!flag) return -1;
    }
    
};

File: 0028.实现strStr()/0028-实现strStr().py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        if not needle:
            return 0
        # if needle in haystack:
        #     return haystack.index(needle)
        for i, char in enumerate(haystack):
            if char == needle[0]:
                if haystack[i:i + len(needle)] == needle:
                    return i
                
        return -1